### PR TITLE
Lighthouse config reload on SIGHUP support

### PR DESCRIPTION
**Pull request by wadey**
This change moves the logic for creating the `LightHouse` structure out of main.go and into `NewLightHouseFromConfig`. It then makes the following config values reload on SIGHUP by registering a reload callback:

- `static_host_map`
- `lighthouse.hosts`
- `lighthouse.interval`
- `punch_back`

This makes it possible to update the static hosts or lighthouses and do a reload, without having to restart the Nebula process.

### CODE REVIEW

File: lighthouse.go [hunk 1]
```diff
@@ -53,6 +55,114 @@ func NewLightHouse(amLighthouse bool, myIp uint32, ips []uint32, interval int, n
...
+	}
+
+	return nil
+}
+
+func (lh *LightHouse) reload(config *Config) {
+	if config.GetBool("lighthouse.am_lighthouse", false) != lh.amLighthouse {
+		l.Warn("Changing lighthouse.am_lighthouse with SIGHUP not supported. Ignored.")
+	}
+	lh.interval = config.GetInt("lighthouse.interval", 10)
```
Comment (nbrownus) [inline lighthouse.go]: Something to note here, if the original config was borked with an insanely high interval a HUP won't help.

File: lighthouse.go [hunk 2]
```diff
@@ -53,6 +55,114 @@ func NewLightHouse(amLighthouse bool, myIp uint32, ips []uint32, interval int, n
...
+			removedStatic = append(removedStatic, vpnIP)
+		}
+	}
+	for vpnIP := range lh.lighthouses {
+		if _, ok := newLh.lighthouses[vpnIP]; !ok {
+			removedLighthouses = append(removedLighthouses, vpnIP)
+		}
+	}
+
+	for _, vpnIP := range removedLighthouses {
```
Comment (nbrownus) [inline lighthouse.go]: This seems highly likely to cause a concurrent read/write access panic in both maps and our current RWMutex is entirely for the `addrMap`, seems we need at least one more now.

This also appears to only remove the diff set and not add any new hosts?

File: main.go [hunk 3]
```diff
@@ -193,70 +190,14 @@ func Main(configPath string, configTest bool, buildVersion string) {
...
-				}
-				lightHouse.AddRemote(ip2int(vpnIp), NewUDPAddr(ip2int(ip), uint16(port)), true)
-			}
-		}
-	}
-
-	err = lightHouse.ValidateLHStaticEntries()
 	if err != nil {
-		l.WithError(err).Error("Lighthouse unreachable")
+		l.Fatalf("Failed to initialize lighthouse handler: %v", err)
```
Comment (nbrownus) [inline main.go]: I'd leave the `WithError` and keep this more helpful error message without the formatting. `Failed to initialize lighthouse`

**Comment by jaysoncena** [comment]
Is it possible to add reloading of `pki.blacklist`?

**Comment by wadey** [comment]
Closed in favor of #649

**Pull request closed without merging**